pilot/pkg/networking/plugin/authn/authentication.go

```diff
@@ -63,7 +64,8 @@ func (Plugin) OnInboundListener(in *plugin.InputParams, mutable *plugin.MutableO
 }
 
 func buildFilter(in *plugin.InputParams, mutable *plugin.MutableObjects) error {
-	applier := factory.NewPolicyApplier(in.Push, in.ServiceInstance)
```

Comment:
You should change this factory function instead. Plugin doesn't need to know which applier (API version) is used.

Reply:
resolved.